internal/databases/sqlserver/log_restore_handler.go

```diff
@@ -56,6 +56,24 @@ func HandleLogRestore(backupName string, untilTS string, dbnames []string, fromn
 	err = runParallel(func(i int) error {
 		dbname := dbnames[i]
 		fromname := fromnames[i]
+		baseURL := getDatabaseBackupURL(backupName, fromname)
+		basePath := getDatabaseBackupPath(backupName, fromname)
+		blobs, err := listBackupBlobs(folder.GetSubFolder(basePath))
+		if err != nil {
+			return err
+		}
+		urls := buildRestoreUrls(baseURL, blobs)
+		backupMetadata, err := ListBackupProperties(db, urls, baseURL)
+		if err != nil {
+			return err
+		}
+		var dbBackupMetadata *BackupProperties
+		for _, dbBackupMetadata = range backupMetadata {
+			if dbBackupMetadata.DatabaseName == fromname {
+				break
+			}
+		}
+		prevBackupFinishdate := dbBackupMetadata.BackupFinishDate
```

Comment:
May it have sense to create helpers like `getDatabaseBackupProperties(backupName, dbName)` & `getLogBackupProperties(logBackupName, dbName)` ? 

BTW, term metadata is already used in wal-g.. it may be confusing. or not )